src/wp-includes/class-wp-customize-manager.php

```diff
@@ -2089,22 +2089,16 @@ public function remove_frameless_preview_messenger_channel() {
 		?>
 		<script>
 		( function() {
-			var urlParser, oldQueryParams, newQueryParams, i;
 			if ( parent !== window ) {
 				return;
 			}
-			urlParser = document.createElement( 'a' );
-			urlParser.href = location.href;
-			oldQueryParams = urlParser.search.substr( 1 ).split( /&/ );
-			newQueryParams = [];
-			for ( i = 0; i < oldQueryParams.length; i += 1 ) {
-				if ( ! /^customize_messenger_channel=/.test( oldQueryParams[ i ] ) ) {
-					newQueryParams.push( oldQueryParams[ i ] );
-				}
-			}
-			urlParser.search = newQueryParams.join( '&' );
-			if ( urlParser.search !== location.search ) {
-				location.replace( urlParser.href );
+			const url = new URL( location.href );
+			const queryParams = url.searchParams;
+
+			if ( queryParams.has( 'customize_messenger_channel' ) ) {
+				queryParams.delete( 'customize_messenger_channel' );
+				url.search = queryParams;
```

Comment:
I'm getting a static analysis warning here since `url.search` is supposed to be a string and `queryParams` is a `URLSearchParams` object:

> ![image](https://github.com/WordPress/wordpress-develop/assets/134745/f6e7b59f-e92c-4610-b5e1-8c82cfe74f31)

I think actually there is no need for this line at all, since `url.searchParams` is updated above and mutating it appears to also update the underlying `URL` object.

In the spec for the URL Standard, it says that the `searchParams` property is `[SameObject]` as the `URL`: https://url.spec.whatwg.org/#url-class

See [definition of `SameObject`](https://webidl.spec.whatwg.org/#SameObject) doesn't really seem to indicate that mutation is propagated up to the "parent" object. Nevertheless, look at [this example](https://url.spec.whatwg.org/#example-constructing-urlsearchparams) in the URL spec:

```js
const url = new URL('https://example.com/?a=b ~');
console.log(url.href);   // "https://example.com/?a=b%20~"
url.searchParams.sort();
console.log(url.href);   // "https://example.com/?a=b+%7E"
```

So you can see that it is intended for mutation to propagate.

See another standalone example: https://codepen.io/westonruter/pen/YzdjzqG?editors=0010

So yeah, unless you see otherwise I think this line should be removed:

```suggestion
```

Reply:
oh wow. I didn't test this but I assumed it cloned the search params since it mentions that property is read-only.

yikes. thanks for checking on this. sounds like we should add a comment explaining that the method calls on `queryParams` modify the URL, or possibly we should eliminate `queryParams` entirely and operate on `url.searchParams.has( … )` and `url.searchParams.delete( … )`?

Reply:
Yes, good idea. Might as well eliminate the `queryParams` variable.

Reply:
Thank you both very much. I just made that change 👍